cmake/clang-format.cmake

```diff
@@ -0,0 +1,50 @@
+# select all files with correct c++ extensions
+file(GLOB_RECURSE
+  ALL_CXX_SOURCE_FILES
+  *.[ch] *.[chi]pp *.[chi]xx *.cppm *.cc *.hh *.ii *.[CHI]
+)
+
+# exclude the docs **for now**
+set(EXCLUDE_REGEX ".*doc[\\|/].*")
+if(${CMAKE_VERSION} VERSION_LESS "3.6.0")
+  foreach(PROPOSED_FILE ${ALL_CXX_SOURCE_FILES})
+    string(REGEX MATCH ${EXCLUDE_REGEX} regex_found ${PROPOSED_FILE})
+    if(regex_found)
+      list(REMOVE_ITEM ${ALL_CXX_SOURCE_FILES} ${PROPOSED_FILE})
+    endif()
+  endforeach()
+else()
+  list(FILTER ALL_CXX_SOURCE_FILES EXCLUDE REGEX ${EXCLUDE_REGEX})
+endif()
+
+find_program(CLANG_FORMAT 
+  NAMES clang-format
+    # backup names
+    clang-format-9.0 clang-format-8.0 clang-format-7.0 clang-format-6.0
+    clang-format-9 clang-format-8 clang-format-7 clang-format-6
+)
+
+# search for version number in clang-format without version number
+if(CLANG_FORMAT)
+  execute_process(
+    COMMAND sh "-c"
+      "${CLANG_FORMAT} --version | grep -oE '[0-9]' | head -n 1" 
```

Comment:
This will break with version 10. Please see the link I posted before, that methods seems pretty robust to me.

Reply:
Sure. Let me create a FindClangFormat.cmake that also works on alpine